From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Karven <karven@atomicmail.io>
Date: Thu, 6 Aug 2026 13:09:28 +0700
Subject: [PATCH] throw error on held slot update


diff --git a/net/minecraft/world/inventory/InventoryMenu.java b/net/minecraft/world/inventory/InventoryMenu.java
index 7a504b015b1abca9de971522332d0df86be2e9bc..bc59445a05be6bfe2544ec9a062dd0e28015e9dd 100644
--- a/net/minecraft/world/inventory/InventoryMenu.java
+++ b/net/minecraft/world/inventory/InventoryMenu.java
@@ -42,7 +42,6 @@ public class InventoryMenu extends AbstractCraftingMenu {
         EMPTY_ARMOR_SLOT_HELMET
     );
     private static final EquipmentSlot[] SLOT_IDS = new EquipmentSlot[]{EquipmentSlot.HEAD, EquipmentSlot.CHEST, EquipmentSlot.LEGS, EquipmentSlot.FEET};
-    private static final org.slf4j.Logger log = org.slf4j.LoggerFactory.getLogger(net.minecraft.world.inventory.InventoryMenu.class);
     public final boolean active;
     private final Player owner;
     private org.bukkit.craftbukkit.inventory.@org.jspecify.annotations.Nullable CraftInventoryView view = null; // CraftBukkit
@@ -212,7 +211,7 @@ public class InventoryMenu extends AbstractCraftingMenu {
         // Otherwise, only resync the hand slot
         final int slot = hand == net.minecraft.world.InteractionHand.MAIN_HAND ? this.owner.getInventory().getSelectedSlot() : Inventory.SLOT_OFFHAND;
         this.forceSlot(this.owner.getInventory(), slot);
-        log.error("slot changed", new RuntimeException()); // Karven - Add debug log
+        throw new RuntimeException("slot updated"); // Karven - throw exception on held slot update
     }
 
     @Override
